includes/Core/Conversion_Tracking/Conversion_Tracking.php

```diff
@@ -96,7 +96,7 @@ public function register() {
 			'wp_enqueue_scripts',
 			function() {
 				// Do nothing if neither Ads nor Analytics snippet has been inserted.
-				if ( ! did_action( 'googlesitekit_ads_init_tag' ) && ! did_action( 'googlesitekit_analytics-4_init_tag' ) ) {
+				if ( ! $this->conversion_tracking_settings->is_conversion_tracking_enabled() || ( ! did_action( 'googlesitekit_ads_init_tag' ) && ! did_action( 'googlesitekit_analytics-4_init_tag' ) ) ) {
```

Comment:
Let's split these onto separate lines as it's quite hard to read as one long line especially with extra parenthesis.

Also, the implementation is getting a bit large to be inline here, let's extract it to a method to keep `register` focused on just registration.